File: CostAccounting.Core/Entities/Core/Income.cs
using System;
using CostAccounting.Core.Entities.Membership;

namespace CostAccounting.Core.Entities.Core
{
    public class Income : Entity<long>
    {
        public const int DescriptionLength = 128;

        public Guid CategoryId { get; set; }
        public Guid UserId { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }

        public Category Category { get; set; }
        public User User { get; set; }
    }
}

File: web/src/app/core/models/income.ts
import { Transaction } from './transaction';
import { Category } from './category';

export class Income implements Transaction {
    id: number;
    amount: number;
    category: Category;
    date: Date;
    description: string;

    constructor(data: any) {
        Object.assign(this, data);

        this.date = new Date(data.date);
    }
}
